### ISSUE TITLE
CMakeTools extension was unable to initialize: AssertionError: Unknown cache entry type

### ISSUE THREAD

**Issue by kriegalex** [vector-of-bool/vscode-cmake-tools#170]
With new v0.9.7, we have access to more error messages. Here is the new one when trying to press F7:

```
CMakeTools extension was unable to initialize: AssertionError: Unknown cache entry type: undefined [See output window for more details]
e.doShow @ messageService.ts:125
e.show @ messageService.ts:104
(anonymous) @ mainThreadMessageService.ts:64
n.Class.derive._oncancel @ winjs.base.raw.js:1638
t.showMessage @ mainThreadMessageService.ts:34
t.$showMessage @ mainThreadMessageService.ts:28
e.handle @ abstractThreadService.ts:34
s @ ipcRemoteCom.ts:275
f @ ipcRemoteCom.ts:231
_combinedTickCallback @ internal/process/next_tick.js:67
_tickCallback @ internal/process/next_tick.js:98
```

**Comment by vector-of-bool** [comment]
What platform/versions are you using? Was this on an already-configured project?

**Comment by kriegalex** [comment]
Yes it was on an already-configured project. Since I got the VS Code 1.13 update, I cannot reproduce the problem. Now, the warning message is:

`Unknown cache entry type: undefined: AssertionError: Unknown cache entry type: undefined`

but the configure stage works now.

**Comment by ytimenkov** [comment]
@kriegalex is it possible to slightly patch the extension to understand what kind of cache entry causes troubles?
 1. Navigate to `%USERPROFILE%\.vscode\extensions\vector-of-bool.cmake-tools-0.10.0\out\src`
 1. Edit `cache.js`
 1. Find line containing "Unknown cache entry type" (~116)
 1. Replace it to something like ``console.assert(type !== undefined, `Unknown cache entry type: ${typename}, ${line}`);`` (Note *type* -> *typename*)
 1. Reload window and find error message in the "Developer Tools" console. (Help -> Toggle Developer Tools).

**Comment by kriegalex** [comment]
Ok, so I have been able to reproduce the problem:

If you start VSCode with the project already configured, it will throw **an error** (the one in my first message).
If you start VSCode with the project build path empty, it will configure it and just leave a warning (as in my second message).

Here is the new message with update cache.js code :

`Unknown cache entry type: AVX":INTERNAL, "HAVE_CXX_ARCH:AVX":INTERNAL=: AssertionError: Unknown cache entry type: AVX":INTERNAL, "HAVE_CXX_ARCH:AVX":INTERNAL=`

**Comment by ytimenkov** [comment]
@kriegalex Thanks for investigating.
CMake has much more sophisticated cache parser than CMake Tools. One noticeable difference is that CMake allows quoting of variables which may contain colons and other symbols:
https://gitlab.kitware.com/cmake/cmake/blob/master/Source/cmState.cxx#L820

I think such entries should be just ignored by CMake Tools because Server Mode should handle it properly.

**Comment by ytimenkov** [comment]
@kriegalex The fix should be available in the latest CMake Tools release.

@vector-of-bool I saw you changed logging to errors when cache entry cant't be parsed.
The reason I made them verbose is that CMake allows colons in variable name. So in certain projects CMake Tools will spam with errors for what CMake itself considers a valid cache entry.

It's a limitation of cache parser of CMake Tools which doesn't allow quoted names. But as I wrote above it is not worth fixing it.
